### PR TITLE
add parameter related interface

### PR DESCRIPTION
**Pull request by czpmango**

### CODE REVIEW

File: client/src/test/java/com/vesoft/nebula/client/graph/net/TestSession.java [hunk 1]
```diff
@@ -173,4 +177,62 @@ public void testReconnectWithMultiServices() {
...
+            // test ping
+            Assert.assertTrue(session.ping());
+            // prepare parameters
+            Map<byte[], Value> paramMap = new HashMap<byte[], Value>();
+            paramMap.put("p1".getBytes(), Value.iVal(3));
+            paramMap.put("p2".getBytes(), Value.bVal(true));
+            paramMap.put("p3".getBytes(), Value.fVal(3.3));
+            // test executeWithParameter interface
+            ResultSet resp = session.executeWithParameter(
+                    "RETURN $p1+1,$p2,$p3", paramMap);
```
Comment (Nicole00) [inline client/src/test/java/com/vesoft/nebula/client/graph/net/TestSession.java:214]: From the perspective of user usage, what’s the difference between parameterized execution and non-parameterized execution?

If users use `executeWithParameter` to execute ngql, them also need to concern the `Value` struct, which may increase the use threshold.

File: client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java [hunk 2]
```diff
@@ -100,60 +100,113 @@ public synchronized ResultSet execute(String stmt) throws
...
-     *                 }
-     *               ],
-     *              "errors":[
-     *                 {
-     *                   "code": 0,
-     *                   "message": ""
-     *                 }
-     *               ]
-     *             }
+     * JSON struct:
```
Comment (Nicole00) [inline client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java:158]: do not format the json struct please.

Comment (czpmango) [reply client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java:158]: ACK.

File: client/src/test/java/com/vesoft/nebula/client/graph/net/TestSession.java (hunk 1 above)
Comment (czpmango) [reply client/src/test/java/com/vesoft/nebula/client/graph/net/TestSession.java]: ACK.

Comment (czpmango) [reply client/src/test/java/com/vesoft/nebula/client/graph/net/TestSession.java]: Fixed.

File: client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java [hunk 3]
```diff
@@ -260,4 +359,77 @@ private boolean retryConnect() {
...
+     */
+    private Value value2Nvalue(Object value) throws
+            UnsupportedOperationException {
+        Value nvalue = new Value();
+        if (value == null) {
+            nvalue.setNVal(NullType.__NULL__);
+        } else if (value instanceof Boolean) {
+            boolean bval = (Boolean)value;
+            nvalue.setBVal(bval);
+        } else if (value instanceof Integer) {
```
Comment (Nicole00) [inline client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java:411]: java  also has types：Byte、Short、Long

File: client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java [hunk 4]
```diff
@@ -260,4 +359,77 @@ private boolean retryConnect() {
...
+            byte[] sval = ((String)value).getBytes();
+            nvalue.setSVal(sval);
+        } else if (value instanceof List) {
+            nvalue.setLVal(list2Nlist((List<Object>)value));
+        } else if (value instanceof Map) {
+            nvalue.setMVal(map2Nmap((Map<String, Object>)value));
+        } else {
+            // unsupport other Value type, use this function carefully
+            throw new UnsupportedOperationException(
+                    "Only support convert boolean/float/int/string/map/list to nebula.Value but was"
```
Comment (Nicole00) [inline client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java:430]: how about Date、DateTime、Time、Timestamp？

File: client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java (hunk 3 above)
Comment (czpmango) [reply client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java:415]: ACK>

File: client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java (hunk 4 above)
Comment (czpmango) [reply client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java:450]: ACK.

File: client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java [hunk 5]
```diff
@@ -260,4 +364,98 @@ private boolean retryConnect() {
...
+     * @return nebula list
+     */
+    private NList list2Nlist(List<Object> list) {
+        NList nlist = new NList();
+        try {
+            for (Object item : list) {
+                nlist.values.add(value2Nvalue(item));
+            }
+        } catch (UnsupportedOperationException e) {
+            throw e;
```
Comment (Nicole00) [inline client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java:380]: if just catch the exception and then throw it, no need to try.

File: client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java [hunk 6]
```diff
@@ -260,4 +364,98 @@ private boolean retryConnect() {
...
+     */
+    private NMap map2Nmap(Map<String, Object> map) {
+        NMap nmap = new NMap();
+        try {
+            for (Map.Entry<String, Object> entry : map.entrySet()) {
+                nmap.kvs.put(entry.getKey().getBytes(), value2Nvalue(entry.getValue()));
+            }
+        } catch (UnsupportedOperationException e) {
+            throw e;
+        }
```
Comment (Nicole00) [inline client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java:398]: ditto

File: client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java [hunk 7]
```diff
@@ -260,4 +364,98 @@ private boolean retryConnect() {
...
+        } else if (value instanceof Date) {
+            return (Value)value;
+        } else if (value instanceof Time) {
+            return (Value)value;
+        } else if (value instanceof Duration) {
+            return (Value)value;
+        } else if (value instanceof DateTime) {
+            return (Value)value;
+        } else if (value instanceof Geography) {
+            return (Value)value;
```
Comment (Nicole00) [inline client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java:452]: Date/Time/Duration/DateTime/Geography is not subclass of Value, cannot directly convert them to Value.

for Date type , use  `nvalue.setDtVal((Date)value)`

Comment (czpmango) [reply client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java:452]: ACK.

File: client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java (hunk 5 above)
Comment (czpmango) [reply client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java:380]: ACK.

File: client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java (hunk 6 above)
Comment (czpmango) [reply client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java:398]: ACK.

**Review by Nicole00 — APPROVED** [review]
LGTM

**Pull request merged**